inferelator_ng/tfa.py

```diff
@@ -0,0 +1,67 @@
+import numpy as np
+import pandas as pd
+from scipy import linalg
+
+class TFA:
+
+    """
+    TFA calculates transcription factor activity using matrix pseudoinverse
+
+        Parameters
+    --------
+    prior: pd.dataframe
+        binary or numeric g by t matrix stating existence of gene-TF interactions. 
+        g--gene, t--TF.
+
+    exp.mat: pd.dataframe
+        normalized expression g by c matrix. g--gene, c--conditions
+
+    exp.mat.halftau: pd.dataframe
+        normalized expression matrix for time series.
+
+    noself=True: boolean
+        If noself (no self interaction) is True, self-regulatory interactions 
+        in prior matrix are set to 0.
+
+    dup_self=True: boolean
+        If dup_slef (duplicate self) is True, TFs that other TFs with the exact same 
+        set of interactions in the prior are kept and will have the same activities
+    """
+
+    def __init__(self, prior, exp_mat, exp_mat_halftau):
+        self.prior = prior
+        self.exp_mat = exp_mat
+        self.exp_mat = exp_mat_halftau
+
+    def tfa(self, noself = True, dup_self = True):
+        tfwt = self.prior.abs().sum(axis = 0) > 0
+        tfwt = tfwt[tfwt].index.tolist()
+
+        if dup_self:
+            # subset of prior matrix whose columns are not all zeros (snz: sum non-zero)
+            prior_snz = self.prior[tfwt]            
+            duplicates = prior_snz.transpose().duplicated(keep=False) # mark duplicates as true
+            dTFs = duplicates[duplicates].index.tolist()
+
+        # find non-duplicated TFs that are also present in target gene list 
+        ndTFs = list(set(self.prior.columns.values.tolist()).difference(dTFs))
+
+        if noself:
```

Comment:
Hi Ren!
Some minor style nits: variable names should avoid capital letters, e.g. `dTFs`, and use underscores where appropriate (`noself` -> `no_self`). 

Echoing @AaronWatters above, one convention that Hammer Lab uses is to avoid abbreviations wherever possible. While this makes the coding a little more verbose, the result is far more readable. E.g. `dTFs` -> `duplicate_tfs`, `ndTFs` -> `non_duplicated_tfs`, `no_self` -> `no_self_interaction`.